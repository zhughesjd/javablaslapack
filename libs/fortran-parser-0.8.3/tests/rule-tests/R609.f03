! Test substring, which is:
!      substring  is  parent-string ( substring-range )
!
! Tested separately are: R610 (parent-string) and R611 (substring-range).
!
! substring tested as part of a designator, which can be part of a primary, 
! so tested as a rhs to an assignment-stmt.
a = my_string(2:3)

end

