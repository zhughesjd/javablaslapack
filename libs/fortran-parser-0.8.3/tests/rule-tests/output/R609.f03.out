*******************************************
args[1]: R609.f03
R1101:main-program__begin:
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R 613:part-ref: id=[@14,283:283='a',<209>,8:0] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 619:section-subscript-list__begin:
R 406:int-literal-constant: digitString=[@20,297:297='2',<10>,8:14] kindParam=null
R 306:literal-constant:
R 701:primary:
R 406:int-literal-constant: digitString=[@22,299:299='3',<10>,8:16] kindParam=null
R 306:literal-constant:
R 701:primary:
R 619:section-subscript: hasLowerBound=true hasUpperBound=true hasStride=false isAmbiguous=false
R 619:section-subscript-list: count=1
R 613:part-ref: id=[@18,287:295='my_string',<209>,8:4] hasSelectionSubscriptList=true hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 208:execution-part:
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=true hasInternalSubprogramPart=false
Parser exiting normally
********************************************
